crates/worker/src/partition/state_machine/mod.rs

```diff
@@ -347,20 +182,17 @@ impl<Codec> StateMachine<Codec> {
     }
 }
 
-pub(crate) struct StateMachineApplyContext<'a, S> {
-    storage: &'a mut S,
+pub(crate) struct StateMachineApplyContext<'a, 'b: 'a> {
+    storage: &'a mut PartitionStoreTransaction<'b>,
```

Comment:
Not sure I agree with replacing with the concrete type here. I think it's still useful to use traits in case we wanna test the individual methods with a simple unit test that doesn't require rocksdb (or if we wanna replace the rocksdb impl with an in memory one later, always for testing)

Reply:
Same comment for essentially all the rest of the file, I feel we should still keep the trait usages, and just use the `Table` traits

Reply:
I guess we can keep it.

Reply:
Well there was a reason not do it because it is super annoying to add the different tables.